tests/tokens.spec.ts

```diff
@@ -0,0 +1,85 @@
+import Asset from "../src/asset";
+import Ledger from "../src/ledger";
+import Tokens, { TokensConfig } from "../src/tokens";
+
+const tokensConfig: TokensConfig = {
+  ethereum: {
+    PAY: {
+      contract: "0xB97048628DB6B661D4C2aA833e95Dbe1A905B280",
+      decimals: 18
+    },
+    TENX: {
+      contract: "0x515bA0a2E286AF10115284F151cF398688A69170",
+      decimals: 18
+    }
+  }
+};
+
+describe("Test Tokens", () => {
+  it("Returns a Tokens instance when an empty object is passed", () => {
+    const tokens = new Tokens({});
+    expect(tokens).toBeDefined();
+  });
+
+  it("Returns a Tokens instance when a correctly formed object is passed", () => {
+    const tokens = new Tokens(tokensConfig);
+    expect(tokens).toBeDefined();
+  });
+
+  it("Throws an error if the passed config has a dupe contract address", () => {
+    expect(() => {
+      // @ts-ignore
+      const tokens = new Tokens({
+        ethereum: {
+          PAY: {
+            contract: "0xB97048628DB6B661D4C2aA833e95Dbe1A905B280",
+            decimals: 18
+          },
+          PAY2: {
+            contract: "0xB97048628DB6B661D4C2aA833e95Dbe1A905B280",
+            decimals: 18
+          }
+        }
+      });
+    }).toThrowError(
+      "Duplicate contract address detected in tokens configuration:"
+    );
+  });
+
+  it("Create an Asset instance when passed a configured contract address", () => {
+    const tokens = new Tokens(tokensConfig);
+    const payAsset = tokens.createAsset(
+      Ledger.Ethereum,
+      "0xB97048628DB6B661D4C2aA833e95Dbe1A905B280"
+    );
+    expect(payAsset).toBeDefined();
+  });
+
+  it("Create an Asset with pre-configured symbol when a configured contract address is passed", () => {
+    const tokens = new Tokens(tokensConfig);
+    const payAsset = tokens.createAsset(
```

Comment:
🔧 
I think it is better to not cast it to `Asset` but define the type as `Asset` as then you will see compile time errors such as forgotten parameters:
```suggestion
    const payAsset: Asset = tokens.createAsset(
```

//Edit// same as in other places.

Reply:
I have to cast to Asset as the `createAsset` function can returned `undefined`.
However, I don't want TS to tell me about it because if it's `undefined` the test will fail as expected.

If I do not cast then I have to do `@ts-ignore` as TS would tell me that `payAsset` could be undefined and hence could not have a `name` property.

Reply:
Fair, thanks for the clarification.